CMakeLists.txt

```diff
@@ -393,11 +397,28 @@ if(RUN_TESTS)
     SourceT/writehero_test.cpp)
 endif()
 
-add_executable(${BIN_TARGET} WIN32 MACOSX_BUNDLE ${devilutionx_SRCS})
-
 # Copy the font to the build directory to it works from the build directory
 file(COPY "Packaging/resources/CharisSILB.ttf" DESTINATION "${CMAKE_CURRENT_BINARY_DIR}")
 
+if (NOT ANDROID)
+  add_executable(${BIN_TARGET}
+    WIN32
+    MACOSX_BUNDLE
+    ${devilutionx_SRCS}
+    SourceX/devilutionx.exe.manifest
+    Packaging/windows/devilutionx.rc
+    Packaging/macOS/AppIcon.icns
+    Packaging/resources/CharisSILB.ttf)
+else()
+  add_library(devilutionx SHARED ${devilutionx_SRCS})
+endif()
+
+if (ANDROID)
+  add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/android-project/SDL2/ ${CMAKE_CURRENT_SOURCE_DIR}/android-project/SDL2)

```

Comment:
Instead of all the ifs here, it'd be better to create INTERFACE target in a dedicate `android_defs.cmake`. Then we won't need any custom Android dependency handling here.

Reply:
Ah, nevermind, I see this is a rather old PR that has SDL sources in-tree.
For reference: https://github.com/xyzz/openmw-android/blob/62694d7ce3db97e558455e9db0a4b4b3d6518d2c/buildscripts/CMakeLists.txt#L270-L287

Reply:
@glebm this PR is specifically about doing it in a way that does not put SDL2 in to the git tree.

If you look at https://github.com/diasurgical/devilutionX-android you can see that only freetype is in tree, which is the last major issue to be solved before we can merge this in.